src/StronglyTypedIds/Templates/NewId/NewId_MongoSerializer.cs

```diff
@@ -0,0 +1,12 @@
+﻿        class TESTIDMongoSerializer : MongoDB.Bson.Serialization.Serializers.SerializerBase<TESTID>
+        {
+            public override TESTID Deserialize(MongoDB.Bson.Serialization.BsonDeserializationContext context, MongoDB.Bson.Serialization.BsonDeserializationArgs args)
+            {
+                return new TESTID(MassTransit.NewId.FromGuid(new System.Guid(context.Reader.ReadString())));
```

Comment:
I don't think you need to go through Guid here? You could use new `NewId(context.Reader.ReadString())` I think 🙂 (similar for `Serialize()`)

Reply:
You have right :) I simplified it